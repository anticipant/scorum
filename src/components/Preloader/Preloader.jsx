import React from 'react';
import './Preloader.scss';
import PropTypes from 'prop-types';

const Preloader = ({ children, isFetching, areImagesShowing = true }) => (
  <React.Fragment>
    {!areImagesShowing || isFetching ? (
      <div className="Preloader">
        <hr />
        <hr />
        <hr />
        <hr />
      </div>
    ) : null }

    {isFetching ? null : children}
  </React.Fragment>
);

Preloader.propTypes = {
  isFetching: PropTypes.bool.isRequired,
  areImagesShowing: PropTypes.bool,
  children: PropTypes.node.isRequired,
};
Preloader.defaultProps = {
  areImagesShowing: true,
};

export default Preloader;
