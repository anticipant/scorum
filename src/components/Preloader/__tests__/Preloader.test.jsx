import React from 'react';
import { shallow } from 'enzyme';
import Preloader from '../Preloader';

describe('Preloader component', () => {
  const props = {
    children: <p>Content Loaded</p>,
    isFetching: false,
  };

  describe('News container initial', () => {
    it('should show that preloader render children with text "Content Loaded"', () => {
      const newsPreloader = shallow(<Preloader {...props} />);

      expect(newsPreloader.find('p').text()).toEqual('Content Loaded');
      expect(newsPreloader).toMatchSnapshot();
    });
  });
  describe('when it is shown Preloader', () => {
    const preloaderHTML = '<div class="Preloader"><hr/><hr/><hr/><hr/></div>';

    it('should show preloader without content (no data received) isFetching = false', () => {
      const nextProps = { ...props, isFetching: true };
      const newsPreloader = shallow(<Preloader {...nextProps} />);

      expect(newsPreloader.html()).toEqual(preloaderHTML);
      expect(newsPreloader).toMatchSnapshot();
    });


    it('should show preloader with content (data received, images not drawn) isFetching = false, areImagesShowing = false', () => {
      const nextProps = { ...props, isFetching: false, areImagesShowing: false };
      const newsPreloader = shallow(<Preloader {...nextProps} />);

      expect(newsPreloader.html()).toEqual(`${preloaderHTML}<p>Content Loaded</p>`);
      expect(newsPreloader).toMatchSnapshot();
    });
  });
});
