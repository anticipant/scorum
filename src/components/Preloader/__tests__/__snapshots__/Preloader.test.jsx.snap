// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preloader component News container initial renders properly 1`] = `
<Fragment>
  <p>
    Content Loaded
  </p>
</Fragment>
`;

exports[`Preloader component News container initial should show that preloader render children with text "Content Loaded" 1`] = `
<Fragment>
  <p>
    Content Loaded
  </p>
</Fragment>
`;

exports[`Preloader component when it is shown Preloader should show preloader with content (data received, images not drawn) isFetching = false, areImagesShowing = false 1`] = `
<Fragment>
  <div
    className="Preloader"
  >
    <hr />
    <hr />
    <hr />
    <hr />
  </div>
  <p>
    Content Loaded
  </p>
</Fragment>
`;

exports[`Preloader component when it is shown Preloader should show preloader without content (no data received) isFetching = false 1`] = `
<Fragment>
  <div
    className="Preloader"
  >
    <hr />
    <hr />
    <hr />
    <hr />
  </div>
</Fragment>
`;
